File: src/com/gnconsult/ieee/Zapatos.java
package com.gnconsult.ieee;

import java.util.HashMap;
import java.util.Scanner;
import java.util.*;

public class Zapatos {

    public static void main(String [] args){
        Scanner sc = new Scanner(System.in);
        System.out.println("Printing the file passed in:");
        String  sn = sc.nextLine();
        int n = Integer.parseInt(sn);

        System.out.println(" n:"+n);
        Map <String,Integer>map1 = new HashMap<String,Integer>();
        Map <String,Integer>tkeys = new HashMap<String,Integer>();
        while(n > 0 && sc.hasNextLine()    ){
            String s1 =sc.nextLine();
            System.err.println("S1:" + s1+" N:"+n);

            if (! s1.trim().equals("")) {
                String tamano = s1.substring(0, 1);
                String lado   = s1.substring(2);
                System.out.println(" Tamano:" + tamano + "  Lado:" + lado);
                String idx = tamano +":"+lado;
                tkeys.put(tamano,1);

                if (map1.containsKey(idx)){
                    int valor = map1.get(idx) +1;
                    map1.put(idx, valor);
                } else{
                    map1.put(idx, 1);
                }
            }
            n--;
        }
        System.out.println(tkeys.keySet());
        System.out.println(map1);

      for (String tamano : tkeys.keySet()){
          if (map1.containsKey(tamano+":L") && map1.containsKey(tamano+":R"))
            System.out.println(tamano+"==>Min "+Math.min(map1.get(tamano+":L"),map1.get(tamano+":R"))) ;
        }
    }

}

File: src/golib/zapatos.go
package main

import (
	"fmt"
)

func main() {
	var N int

	fmt.Println("Inicio Zapatos Go")
	fmt.Scanf("%d", &N)
	fmt.Println("N:", N)
	//Map<String,Integer> map1 = new HashMap<String,Integer>();
	//Map <String,Integer>tkeys = new HashMap<String,Integer>();
	var map1 map[string]int
	map1 = make(map[string]int)

	var tkeys map[string]int
	tkeys = make(map[string]int)

	for i := 0; i < N; i++ {
		var tamano string
		var lado string
		fmt.Scanf("%s %s\n", &tamano, &lado)
		//fmt.Println("tamano:",tamano, "   lado:", lado)
		var idx = tamano + "." + lado

		tkeys[tamano] = 1
		var valor = map1[idx]
		if valor == 0 {
			map1[idx] = 1
			//fmt.Println("valor1:",valor)
		} else {
			map1[idx] = map1[idx] + 1
		}
	}
	fmt.Println("tkeys:", tkeys)
	fmt.Println("map1:", map1)
	for key := range tkeys {
		fmt.Println("tamano:", key, "  R:", map1[key+".R"], "L:", map1[key+".L"], "", Min(map1[key+".R"], map1[key+".L"]))
	}
}

func Min(x, y int) int {
	if x > y {
		return y
	}
	return x
}
